src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/ManageUsersService.kt

```diff
@@ -81,4 +84,12 @@ class ManageUsersService(
       manageUsersApiClient.getUserDetails(userName).block(apiTimeout)?.fullName ?: userNameIfNotAvailable
     }
   }
+
+  fun getFullNameFromActionedBy(actionedByDto: ActionedByDto): String {
+    return when (actionedByDto.userType) {
+      UserType.STAFF -> manageUsersApiClient.getUserDetails(actionedByDto.userName!!).block(apiTimeout)?.fullName ?: NOT_KNOWN
```

Comment:
should we use getUserFullName method instead (line 80)?

Reply:
I've opted away from this as it's used in a few other places that don't have access to the ActionedByDto. They pass directly a String. Avoiding a big refactor by introducing a new method.